katib-ui-rock/rockcraft.yaml

```diff
@@ -19,36 +19,86 @@ platforms:
   amd64:
 
 parts:
-  katib-ui:
+  frontend-lib:
+    plugin: nil
+    source: https://github.com/kubeflow/kubeflow
+    source-type: git
+    # Set source-tag here to the commit cited in https://github.com/kubeflow/katib/blob/master/pkg/ui/v1beta1/frontend/COMMIT
+    # for this Katib release
+    # Note: Katib 0.16.0 cannot build with Kubeflow v1.8-branch because Kubeflow refactored this lib
+    source-commit: 046c6d3c8301a95a2809c0d927bbbf72638dd639
+    # Omit source-depth because we're cloning a commit
+    # source-depth: 1
+    build-environment:
+      # suppress analytics collection
+      - NG_CLI_ANALYTICS: false
+    build-snaps:
+      - node/12/stable
+    override-build: |
+      # Clean the build dir so we don't accidentally use files we didn't intend
+      rm -rf ./*
+
+      # Pull only the files we need for the build from src, and build
+      cp $CRAFT_PART_SRC/components/crud-web-apps/common/frontend/kubeflow-common-lib/package*.json ./
+      npm ci
+      cp -r $CRAFT_PART_SRC/components/crud-web-apps/common/frontend/kubeflow-common-lib/* ./
+      npm run build
+      
+    override-stage: |
+      mkdir -p $CRAFT_STAGE/frontend-lib/dist
+      cp -r $CRAFT_PART_BUILD/dist/kubeflow $CRAFT_STAGE/frontend-lib/dist/kubeflow
+
+  frontend:
+    after: [frontend-lib]
+    plugin: nil
+    source: https://github.com/kubeflow/katib
+    source-type: git
+    source-tag: v0.16.0
+    source-depth: 1
+    build-snaps:
+      - node/12/stable
+    override-build: |
+      # Clean the build dir so we don't accidentally use files we didn't intend
+      rm -rf ./*
+
+      # Pull only the files we need for the build from src
+      cp $CRAFT_PART_SRC/pkg/ui/v1beta1/frontend/package*.json ./
+      npm ci
+      cp -r $CRAFT_STAGE/frontend-lib/dist/kubeflow ./node_modules/kubeflow
+      cp -r $CRAFT_PART_SRC/pkg/ui/v1beta1/frontend/* .
+      npm run build:prod
+
+    override-prime: |
+      mkdir -p $CRAFT_PRIME/app/build
+      cp -r $CRAFT_PART_BUILD/dist/static $CRAFT_PRIME/app/build/static
+
+  backend:
     plugin: go
     source: https://github.com/kubeflow/katib
     source-type: git
-    source-tag: v0.15.0
+    source-tag: v0.16.0
     build-snaps:
-      - go
-      - curl
-    build-packages:
-      - npm
-    stage-packages:
-      - libc6_libs
+      - go/1.20/stable
```

Comment:
Why 1.20? Maybe it would make sense to document how did you decide on the go version since this is not straight forward form looking at the Dockerfile (so we can be cautious during updates it in the future)

Reply:
I have no idea - tbh I'm not even sure why I picked 1.20.  The upstream dockerfile is not specific either

Reply:
Got it [here](https://github.com/kubeflow/katib/blob/v0.16.0/go.mod#L3). (looked at the [CI](https://github.com/kubeflow/katib/blob/v0.16.0/.github/workflows/test-go.yaml#L47)). We can keep 1.20 with a comment pointing to that file